xbmc/video/VideoDatabase.cpp

```diff
@@ -4132,6 +4132,7 @@ CVideoInfoTag CVideoDatabase::GetDetailsForTvShow(const dbiplus::sql_record* con
     item->SetProperty("numepisodes", details.m_iEpisode); // will be changed later to reflect watchmode setting
     item->SetProperty("watchedepisodes", details.GetPlayCount());
     item->SetProperty("unwatchedepisodes", details.m_iEpisode - details.GetPlayCount());
+    item->SetProperty("watchedepisodepercent", (details.GetPlayCount() * 100 / details.m_iEpisode));
```

Comment:
should this be generally casted to int, since CVariant can take floats and doubles? just for safety, rounding issues etc.

Reply:
disregard, shouldn‘t matter as only int‘s involved

Reply:
the division is inverted (multiplication by 100) to avoid precisely that otherwise we'd get a 0% if you watched 1 episode out of 3 since only ints are involved

Reply:
One thing to consider is that we use floor division here. If there are 1000 episodes, and the user watches 9, it'll say 0%, even though it's closer to 1%. However, I think this is OK, becuase we actually *want* this behavior at the other end - the user doesnt want to see 100% if they've watched 999/1000, they want to see 99% to show that there's still one left. So I think flooring is OK for this property.